runtime/autoload/provider/clipboard.vim

```diff
@@ -4,43 +4,88 @@
 let s:copy = {}
 let s:paste = {}
 
+" when caching is enabled, store the jobid of the xclip/xsel process
+" keeping ownership of the selection, so we know how long the cache is valid
+let s:selection = { 'owner': 0, 'data': [] }
+
+function! s:selection.on_exit(jobid, data, event)
+  " At this point this nvim instance might already have launched
+  " a new provider instance. Don't drop ownership in this case.
+  if self.owner == a:jobid
+    let self.owner = 0
+  endif
+endfunction
+
+let s:selections = { '*': s:selection, '+': copy(s:selection)}
+
 function! s:try_cmd(cmd, ...)
-  let out = a:0 ? systemlist(a:cmd, a:1, 1) : systemlist(a:cmd, [''], 1)
+  let argv = split(a:cmd, " ")
+  let out = a:0 ? systemlist(argv, a:1, 1) : systemlist(argv, [''], 1)
   if v:shell_error
-    echo "clipboard: error: ".(len(out) ? out[0] : '')
-    return ''
+    echohl WarningMsg
+      echo "clipboard: error: ".(len(out) ? out[0] : '')
+    echohl None
+    return 0
   endif
   return out
 endfunction
 
+let s:cache_enabled = 1
 if executable('pbcopy')
   let s:copy['+'] = 'pbcopy'
   let s:paste['+'] = 'pbpaste'
   let s:copy['*'] = s:copy['+']
   let s:paste['*'] = s:paste['+']
+  let s:cache_enabled = 0
 elseif executable('xclip')
-  let s:copy['+'] = 'xclip -i -selection clipboard'
+  let s:copy['+'] = 'xclip -quiet -i -selection clipboard'
   let s:paste['+'] = 'xclip -o -selection clipboard'
-  let s:copy['*'] = 'xclip -i -selection primary'
+  let s:copy['*'] = 'xclip -quiet -i -selection primary'
   let s:paste['*'] = 'xclip -o -selection primary'
 elseif executable('xsel')
-  let s:copy['+'] = 'xsel -i -b'
+  let s:copy['+'] = 'xsel --nodetach -i -b'
   let s:paste['+'] = 'xsel -o -b'
-  let s:copy['*'] = 'xsel -i -p'
+  let s:copy['*'] = 'xsel --nodetach -i -p'
   let s:paste['*'] = 'xsel -o -p'
 else
   echom 'clipboard: No shell command for communicating with the clipboard found.'
+  echom 'clipboard: see :help nvim-clipboard'
```

Comment:
I'm strongly opposed to these "see :help foo" messages we are adding to the error messages. We should prefix with EXX error numbers instead. 

We should not optimize error messages for new users (you're only *new* for a short period of time), because regular users suffer this message spam forever thereafter. Yes, we do need to help new users, but let's find a different way.